v2/yarpcjson/outbound.go

```diff
@@ -63,25 +64,41 @@ func (c Client) Call(ctx context.Context, procedure string, reqBody interface{},
 		return yarpcencoding.RequestBodyEncodeError(&req, err)
 	}
 
-	res, resBuf, appErr := c.c.Unary.Call(ctx, &req, yarpc.NewBufferBytes(encoded))
-	if res == nil {
-		return appErr
+	res, resBuf, err := c.c.Unary.Call(ctx, &req, yarpc.NewBufferBytes(encoded))
+	if err != nil {
+		return err
 	}
 
-	// we want to return the appErr if it exists as this is what
-	// the previous behavior was so we deprioritize this error
-	var decodeErr error
 	if _, err = call.ReadFromResponse(ctx, res); err != nil {
-		decodeErr = err
+		return err
 	}
-	if resBuf != nil {
-		if err := json.NewDecoder(resBuf).Decode(resBodyOut); err != nil && decodeErr == nil {
-			decodeErr = yarpcencoding.ResponseBodyDecodeError(&req, err)
+
+	if res.ApplicationErrorInfo != nil {
```

Comment:
nit: Since we're accessing `res.ApplicationErrorInfo` a lot in this branch, we can initialize a separate, more readable variable for it:
```
if info := res.ApplicationErrorInfo; info != nil { ... }
```